src/main/java/io/github/klsmith/budgettracker/money/budget/Sql2oBudgetDao.java

```diff
@@ -0,0 +1,140 @@
+package io.github.klsmith.budgettracker.money.budget;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.sql2o.Connection;
+import org.sql2o.Sql2o;
+
+import io.github.klsmith.budgettracker.dao.DaoException;
+import io.github.klsmith.budgettracker.sql2o.Sql2oDao;
+import io.github.klsmith.budgettracker.tag.Sql2oTagDao;
+import io.github.klsmith.budgettracker.tag.Tag;
+
+/**
+ * Implementation of {@link BudgetDao} using Sql2o.
+ */
+public class Sql2oBudgetDao extends Sql2oDao implements BudgetDao {
+
+    private final Sql2oTagDao tagDao;
+
+    /**
+     * Construct using Sql2o object and Sql2oTagDao.
+     */
+    public Sql2oBudgetDao(Sql2o sql2o, Sql2oTagDao tagDao) {
+        super(sql2o);
+        this.tagDao = tagDao;
+    }
+
+    @Override
+    public Budget create(Budget budgetData) {
+        return transaction(connection -> create(connection, budgetData));
```

Comment:
This line seems a bit cryptic to me, but I'm unsure of how to improve it just yet.

Reply:
It needs a lot of refactoring. I completely agree with you, but it wouldn't be a small change to fix it.